wasmer/config.go

```diff
@@ -166,6 +167,36 @@ func (self *Config) UseCraneliftCompiler() *Config {
 	return self
 }
 
+var opCodeMap map[Opcode]uint32 = nil
+
+//export metering_delegate
+func metering_delegate(op C.wasmer_parser_operator_t) C.uint64_t {
+	// a simple alogorithm for now just map from opcode to cost directly
+	// all the responsibility is placed on the caller of PushMeteringMiddleware
+	v, b := opCodeMap[Opcode(op)]
+	if !b {
+		return 0 // no value means no cost
+	}
+	return C.uint64_t(v)
+}
+
+// PushMeteringMiddleware allows the middleware metering to be engaged on a map of opcode to cost
+//   config := NewConfig()
+//	 opmap := map[uint32]uint32{
+//		13: 1,
+//		25: 1,
+//		99: 4,
```

Comment:
Can you update the docs using the Opcode instead of the integers?

Reply:
Silly me forgot!  Sorry.

Reply:
(also probably `map[Opcode]uint32` instead of `map[uint32]uint32`)